src/metabase/events/view_log.clj

```diff
@@ -3,63 +3,101 @@
   (:require
    [metabase.api.common :as api]
    [metabase.events :as events]
-   [metabase.models.setting :as setting :refer [defsetting]]
+   [metabase.models.audit-log :as audit-log]
+   [metabase.models.query.permissions :as query-perms]
    [metabase.util :as u]
-   [metabase.util.i18n :as i18n :refer [deferred-tru]]
    [metabase.util.log :as log]
    [methodical.core :as m]
    [toucan2.core :as t2]))
 
-(defn record-view!
+(defn- record-views!
   "Simple base function for recording a view of a given `model` and `model-id` by a certain `user`."
-  [model model-id user-id metadata]
-  (t2/insert! :model/ViewLog
-              :user_id  user-id
-              :model    (u/lower-case-en model)
-              :model_id model-id
-              :metadata metadata))
+  [view-or-views]
+  (t2/insert! :model/ViewLog view-or-views))
 
-(derive ::read-event :metabase/event)
-(derive :event/card-read ::read-event)
-(derive :event/dashboard-read ::read-event)
-(derive :event/table-read ::read-event)
+(defn- generate-view
+  "Generates a view, given an event map."
+  [{:keys [object user-id has-access]
+    :or   {has-access true}}]
+  {:model      (u/lower-case-en (audit-log/model-name object))
```

Comment:
I'm standardizing model names as being lowercase in `view_log` and `recent_views` even though they're upper-case in `audit_log`. I wish they could be standardized as upper-case for all three tables, but enough code currently relies on them being lowercase for views, that it's not worth breaking anything.